### PR TITLE
WT-12469 Remove log-based incremental backup tests/examples

### PR DESCRIPTION
**Pull request by korteland**
The diff for `ex_backup.c` is confusing, but all I did was rename `ex_backup_block.c` to `ex_backup.c`.

There are also a number of doc changes in this PR, this is temporary so that `s_all` will still run. The documentation for log-based incremental backup is getting removed properly in WT-12471.

The test/format changes are something I could use some feedback on. The trouble is upgrade/downgrade testing. Old branches could still generate log-based incremental backup configs. The way we've fixed that so far is to add `foo=0 >> $file_name` in `compatibility_test_for_releases.sh`. But that doesn't work for the scenario where we remove a choice, like we are here. So it seems best to (1) allow older versions to generate configs with log-based incremental backup, (2) understand the config, and (3) do nothing with it. If people agree about this, I'll make a ticket to remove the compatibility code in a future version.

### CODE REVIEW

**Comment by sueloverso** [comment]
@korteland these changes all for current code to continue to generate {{backup.incremental=log}} because it remains a choice in {{config.sh}}. It is clearly now a boolean. It would be preferable for format to generate {{backup.incremental=0|1}}. I can live with {{backup.incremental=block|off}}. I don't mind your fix for the compatibility of {{foo=0}} or {{foo=off}} if necessary. It also means that the runners in {{failure_configs}} need editing. The rest of the changes are okay with me.

**Review by tod-johnson-mongodb — APPROVED** [review]
LGTM

File: examples/c/ex_backup.c [hunk 1]
```diff
@@ -153,80 +254,188 @@ take_full_backup(WT_SESSION *session, int i)
...
+    count = 0;
+    alloc = FLIST_INIT;
+    flist = calloc(alloc, sizeof(FILELIST));
+    testutil_assert(flist != NULL);
     while ((ret = cursor->next(cursor)) == 0) {
         error_check(cursor->get_key(cursor, &filename));
+        error_check(process_file(&flist, &count, &alloc, filename));
+
+        /*
+         * If it is a log file, prepend the path for cp.
```
Comment (ruby-oujo) [inline examples/c/ex_backup.c:273]: since it's a comment, this should be copy not cp.

File: examples/c/ex_backup.c [hunk 2]
```diff
@@ -254,6 +467,37 @@ main(int argc, char *argv[])
         error_check(compare_backups(i));
     }
 
+    printf("Close and reopen the connection\n");
+    /*
+     * Close and reopen the connection to illustrate the durability of id information.
```
Comment (ruby-oujo) [inline examples/c/ex_backup.c:472]: np: id -> ID

**Review by ruby-oujo — APPROVED** [review]
LGTM!

File: examples/c/ex_backup.c (hunk 1 above)
Comment (korteland) [reply examples/c/ex_backup.c:273]: I don't agree with this one - the `cp` refers to the specific `testutil_system` call below where we call `cp`.

**Comment by korteland** [comment]
@sueloverso Thanks for pointing that out. I've fixed up the test case generation. I think we'll have to live with `backup.incremental=block|off`, since that's the only way I can keep compatibility with the `log` option that the upgrade/downgrade tests will generate.

I've also checked that log-based incremental backup was not important for the two failure configs that needed tweaking. New code is in 931885b, let me know what you think.

**Review by sueloverso — APPROVED** [review]
LGTM. The format changes are nicely contained.

**Pull request merged**